I disabled the test for `open-ping` because the test runner was failing to run the traceroute command in the shell:
```
FAILED: open-ping Traceback (most recent call last):
...
  File "/home/runner/work/visidata/visidata/visidata/features/ping.py", line 103, in send_trace
    sh.traceroute('--sim-queries=1',
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/sh.py", line 3657, in __getattr__
    return self.__env[name]
  File "/opt/hostedtoolcache/Python/3.10.13/x64/lib/python3.10/site-packages/sh.py", line 3438, in __getitem__
    raise CommandNotFound(k)
sh.CommandNotFound: traceroute
1/440 commands had errors
FAILED
```

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -43,6 +43,9 @@ jobs:
     - name: Install lxml dependencies
       run: sudo apt install -y libxml2-dev libxslt-dev
 
+    - name: Install ping dependencies
+      run: sudo apt install -y traceroute
+
     - name: Install optional dependencies
       run: |
         pip3 install -r requirements.txt

--- a/visidata/features/ping.py
+++ b/visidata/features/ping.py
@@ -10,9 +10,14 @@ from visidata import vd, VisiData, BaseSheet, Sheet, Column, AttrColumn, Progres
 @VisiData.api
 def new_ping(vd, p):
     'Open a sheet with the round-trip time for each hop along the path to the given host.'
-    pingsheet = PingSheet(p.given, source=p.given)
-    return PingStatsSheet("traceroute_"+pingsheet.name, source=pingsheet)
+    return makePingStats(p.given)
 
+# open_ping() is called when a local file exists matching the path p, otherwise new_ping() is called
+vd.open_ping = vd.new_ping
+
+def makePingStats(ip):
+    pingsheet = PingSheet(ip, source=ip)
+    return PingStatsSheet("traceroute_"+pingsheet.name, source=pingsheet)
 
 class PingStatsSheet(Sheet):
     help='''# ping/traceroute
@@ -71,7 +76,7 @@ class PingSheet(Sheet):
     def ping_error(self, ip, data):
         if ip in self.sources:
             self.sources.remove(ip)
-            vd.warning("%s removed: %s" % (ip, data))
+            vd.warning("%s removed: %s" % (ip, data.rstrip()))
 
     def update_traces(self, row, ip):
         rtes = self.routes.get(ip)
@@ -145,7 +150,11 @@ vd.option('ping_count', 3, 'send this many pings to each host', sheettype=PingSh
 vd.option('ping_interval', 0.1, 'wait between ping rounds, in seconds', sheettype=PingSheet)
 
 PingSheet.options.null_value = False
-BaseSheet.addCommand('', 'open-ping', 'vd.push(openSource(input("ping: ", type="hostip"), filetype="ping"))', 'open sheet to ping input IP Address')
+BaseSheet.addCommand('', 'open-ping', 'vd.push(makePingStats(vd.input("ping: ", type="hostip")))', 'open sheet to ping input IP Address')
+
+vd.addGlobals(
+    makePingStats=makePingStats,
+)
 
 vd.addMenuItems('''
     System > Ping IP/hostname > open-ping

--- a/visidata/features/ping.py
+++ b/visidata/features/ping.py
@@ -36,7 +36,7 @@ This sheet runs `traceroute` to generate intermediate hops, then runs `ping` aga
     ]
     def loader(self):
         sheet = self.source
-        sheet.ensureLoaded()
+        vd.sync(sheet.ensureLoaded())
 
         self.rows = sheet.columns
 

--- a/visidata/tests/test_commands.py
+++ b/visidata/tests/test_commands.py
@@ -32,6 +32,7 @@ nonTested = (
         'menu',
         'sysopen',
         'open-memusage',
+        'open-ping',
         )
 
 def isTestableCommand(longname, cmdlist):

--- a/visidata/tests/test_commands.py
+++ b/visidata/tests/test_commands.py
@@ -32,7 +32,6 @@ nonTested = (
         'menu',
         'sysopen',
         'open-memusage',
-        'open-ping',
         )
 
 def isTestableCommand(longname, cmdlist):
@@ -69,6 +68,7 @@ inputLines = { 'save-sheet': 'jetsam.csv',  # save to some tmp file
                  'setcol-incr-step': '2',
                  'setcol-iter': 'range(1, 100)',
                  'setcol-format-enum': '1=cat',
+                 'open-ping': 'github.com',
                  'setcol-input': '5',
                  'show-expr': 'OrderDate',
                  'setcol-expr': 'OrderDate',

